File: 400/459 - Repeated Substring Pattern.c
#include <string.h>

bool repeatedSubstringPattern(char* s) {
    int len, i, j, k, flag;
    
    len = strlen(s);
    
    for (i = len / 2; i >= 1; --i) {
        if (! (len % i)) {
            flag = 0;
            for (j = 0; j < i; ++j) {
                for (k = j; k + i < len; ++k) {
                    if (s[k] != s[k + i]) {
                        flag = 1;
                        break;
                    }
                }
                
                if (flag)
                    break;
                else
                    return 1;
            }
        }
    }
    
    return 0;
}

File: csharp/459.repeated-substring-pattern.cs
/*
 * @lc app=leetcode id=459 lang=csharp
 *
 * [459] Repeated Substring Pattern
 */
public class Solution {
    public bool RepeatedSubstringPattern(string s) {
        int n = s.Length;
        for (int i = 1; i <= n / 2; ++i) {
            if (n % i != 0) {
                continue;
            }
            string sub = s.Substring(0, i);
            bool flag = true;
            for (int j = i; j + i <= n; j += i) {
                if (sub != s.Substring(j, i)) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                return true;
            }
        }
        return false;
    }
}

